src/__tests__/Document.test.ts

```diff
@@ -856,7 +856,133 @@ describe('validate', () => {
 			});
 			const document = new DocumentSubclass(schema, { data: { prop1: 'foo' } });
 
-			const expected = new Map([['prop1', 'Test error message']]);
+			const expected = new Map([['prop1', ['Test error message']]]);
+			expect(await document.validate()).toEqual(expected);
+		});
+
+		test('return error at property & unravel nested errors if schemaType validation fails', async () => {
+			const passingEmbeddedObjectSchema: SchemaDefinition = {
+				prop1: [{ type: 'number', path: '7', required: true }],
+				prop2: { type: 'number', path: '8', required: true },
+			};
+			const failingEmbeddedObjectSchema: SchemaDefinition = {
+				prop1: [{ type: 'number', path: '9', required: true }],
+				prop2: { type: 'number', path: '10', required: true },
+			};
+
+			const passingEmbeddedObjectDocumentArraySchema: SchemaDefinition = {
+				prop1: [{ type: 'string', path: '11', required: true }],
+				prop2: { type: 'number', path: '12', dbDecimals: 2, required: true },
+			};
+			const failingEmbeddedObjectDocumentArraySchema: SchemaDefinition = {
+				prop1: [{ type: 'string', path: '13', required: true }],
+				prop2: { type: 'number', path: '14', dbDecimals: 2, required: true },
+			};
+
+			const passingNestedSchema = new Schema({
+				prop1: { type: 'string', path: '15', required: true },
+				prop2: { type: 'number', path: '16', dbDecimals: 2, required: true },
+			});
+			const failingNestedSchema = new Schema({
+				prop1: [{ type: 'string', path: '17', required: true }],
+				prop2: { type: 'number', path: '18', dbDecimals: 2, required: true },
+			});
+
+			const passingNestedSchemaDocumentArraySchema = new Schema({
```

Comment:
These are being referred to as "nested", but I wouldn't think of them as such. You can eliminate the `NestedSchema` part of the name and make it easier to read and understand.